app/models/pseud.rb

```diff
@@ -259,28 +249,37 @@ def self.parse_byline(byline, options = {})
   # Takes a comma-separated list of bylines
   # Returns a hash containing an array of pseuds and an array of bylines that couldn't be found
   def self.parse_bylines(list, options = {})
-    valid_pseuds, ambiguous_pseuds, failures = [], {}, []
+    valid_pseuds = []
+    ambiguous_pseuds = {}
+    failures = []
+    banned_pseuds = []
     bylines = list.split ","
     for byline in bylines
       pseuds = Pseud.parse_byline(byline, options)
-      banned_pseuds = pseuds.select { |pseud| pseud.user.banned? || pseud.user.suspended? }
-      if banned_pseuds.present?
-        pseuds = pseuds - banned_pseuds
-        banned_pseuds = banned_pseuds.map(&:byline)
+      remove_pseuds = false
+      if pseuds.empty?
+        failures << byline.strip
+        next
       end
-      if pseuds.length == 1
-        valid_pseuds << pseuds.first
-      elsif pseuds.length > 1
+      if pseuds.size > 1
         ambiguous_pseuds[pseuds.first.name] = pseuds
-      else
-        failures << byline.strip
+        remove_pseuds = options[:remove_ambiguous]
```

Comment:
Where is this option used?

Reply:
It's not used, but I'm not certain I should remove it. The changes in `Pseud.parse_bylines` don't really fix anything (except for a few bugs involving banned pseuds). The bug that `parse_bylines` sort of addresses is the issue with adding ambiguous names as owners of a TagSet ([AO3-4867](https://otwarchive.atlassian.net/browse/AO3-4867)). But it doesn't actually fix AO3-4867, it just causes a different bad behavior: instead of being unable to add ambiguous pseuds, it'll add every user who has a pseud with that name.

In general, I think it's important to be able to exclude ambiguous pseuds from the list of valid pseuds. IMHO, they should always be removed, and the OwnedTagSet bug should be resolved by calling `parse_bylines` with the `assume_matching_login` option. (Or by writing more complex code to allow users to resolve the ambiguity.)

But IIRC, the change to `parse_bylines` was one of the things that @zz9pzza wanted me to include, so I'm not the right person to ask.

Reply:
> instead of being unable to add ambiguous pseuds, it'll add every user who has a pseud with that name.

The new bad behavior sounds worse than the old one.

Reply:
**Infinitely** worse and something we absolutely cannot deploy. The old bad behavior was annoying, but we could tell folks to work around it by entering `pseud (username)`. The new bad behavior runs the risk of the wrong person getting access to a tag set and deleting it.

Reply:
I'll set it to always remove ambiguous pseuds.

Reply:
Looks good, thanks!